src/GameState/Marble.tsx

```diff
@@ -23,16 +24,21 @@ export const Marble = (props: any) => {
   );
 
   const handleClickMarble = async () => {
-    props.setMarbleClicked({
-      x: props.cellIndex,
-      y: props.rowIndex,
-      value: props.value,
-      isExit: false,
-    });
-    props.setMarbleClickedCoordinates({
-      x: props.cellIndex,
-      y: props.rowIndex,
-    });
+    clicked = !clicked;
+    if (clicked) {
+      props.setMarbleClicked({
+        x: props.cellIndex,
+        y: props.rowIndex,
+        value: props.value,
+        isExit: false,
+      });
+      props.setMarbleClickedCoordinates({
+        x: props.cellIndex,
+        y: props.rowIndex,
+      });
+    } else {
+      props.setMarbleClickedCoordinates(undefined);
```

Comment:
It's not a good idea to use `undefined` as the empty state. It's better to use `null` because we are able to make the difference between something which has not been declared and something voluntary set to empty.